### PR TITLE
Generate crops in Blender verifier

### PR DESCRIPTION
**Pull request by badb**

### CODE REVIEW

**Review by Elfoniok — APPROVED** [review]

File: apps/blender/task/verifier.py [hunk 1]
```diff
@@ -72,5 +75,46 @@ def _render_crops(self, subtask_info, resources, num_crops=NUM_CROPS,
...
+        ctd['extra_data']['outfilebasename'] = \
+            "ref_" + subtask_info['outfilebasename']
+        ctd['extra_data']['script_src'] = script_src
+        ctd['deadline'] = timeout_to_deadline(subtask_info['subtask_timeout'])
+        return ctd
+
+    @staticmethod
+    def _crop_rendered(results, time_spend):
+        logger.info("Crop for verification rendered. Time spent: {}, "
+                    "results: {}".format(time_spend, results))
```
Comment (jiivan) [inline apps/blender/task/verifier.py]: `[logging-format-interpolation] Use % formatting in logging functions and pass the % parameters as arguments`
https://gist.github.com/jiivan/eedb505574ccb2503af48fa2e3043036

File: apps/blender/task/verifier.py [hunk 2]
```diff
@@ -72,5 +75,46 @@ def _render_crops(self, subtask_info, resources, num_crops=NUM_CROPS,
...
+
+    @staticmethod
+    def _crop_render_failure(error):
+        logger.info("Crop for verification render failure {}".format(error))
```
Comment (jiivan) [inline apps/blender/task/verifier.py]: `[logging-format-interpolation] Use % formatting in logging functions and pass the % parameters as arguments`

File: apps/blender/task/verifier.py [hunk 3]
```diff
@@ -72,5 +75,46 @@ def _render_crops(self, subtask_info, resources, num_crops=NUM_CROPS,
...
+        minx, maxx, miny, maxy = crop
+
+        script_src = generate_blender_crop_file(
+            resolution=(subtask_info['res_x'], subtask_info['res_y']),
+            borders_x=(minx, maxx),
+            borders_y=(miny, maxy),
+            use_compositing=False
+        )
+        ctd = self._generate_ctd(subtask_info, script_src)
+        # FIXME (root path)
```
Comment (jiivan) [inline apps/blender/task/verifier.py]: What does it mean? Is there an issue for that?

File: apps/blender/task/verifier.py [hunk 4]
```diff
@@ -61,8 +66,6 @@ def _verify_imgs(self, subtask_info, results, reference_data, resources):
             return False
         return True
 
-    # FIXME remove pylint-disable
-    # pylint: disable=unused-argument
```
Comment (jiivan) [inline apps/blender/task/verifier.py]: pylint still reports unused argument
`apps/blender/task/verifier.py|69 col 43 warning| [unused-argument] Unused argument 'resources'`

File: apps/blender/task/blenderrendertask.py [hunk 5]
```diff
@@ -420,6 +420,9 @@ def query_extra_data(self, perf_index, num_cores=0, node_id=None, node_name=None
         self.subtasks_given[hash]['use_frames'] = self.use_frames
         self.subtasks_given[hash]['all_frames'] = self.frames
         self.subtasks_given[hash]['crop_window'] = (0.0, 1.0, min_y, max_y)
+        self.subtasks_given[hash]['subtask_timeout'] = \
+            self.header.subtask_timeout
+        self.subtasks_given[hash]['tmp_dir'] = self.tmp_dir  # FIXME
```
Comment (jiivan) [inline apps/blender/task/blenderrendertask.py]: What's here to fix? Is there an issue for that?

**Review by jiivan — CHANGES_REQUESTED** [review]
LGTM minus easily fixable issues. Generally I would appreciate more docstrings and comments in whole verifier class.

File: apps/blender/task/blenderrendertask.py (hunk 5 above)
Comment (badb) [reply apps/blender/task/blenderrendertask.py]: https://github.com/golemfactory/golem/issues/1955

File: apps/blender/task/verifier.py (hunk 3 above)
Comment (badb) [reply apps/blender/task/verifier.py]: #1955

File: apps/blender/task/blenderrendertask.py (hunk 5 above)
Comment (jiivan) [reply apps/blender/task/blenderrendertask.py]: Could put a reference to this issue in `# FIXME` comment?

File: apps/blender/task/verifier.py (hunk 3 above)
Comment (jiivan) [reply apps/blender/task/verifier.py]: Could put a reference to this issue in `# FIXME` comment?

**Review by jiivan — APPROVED** [review]

**Pull request merged**